packages/react-native/ReactCommon/react/renderer/graphics/ValueUnit.cpp

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "ValueUnit.h"
+
+#include <react/renderer/graphics/Float.h>
+
+namespace facebook::react {
+
+ValueUnit ValueUnit::getValueUnitFromRawValue(const RawValue& value) {
+  if (value.hasType<Float>()) {
+    auto number = (float)value;
+    return ValueUnit(number, UnitType::Point);
+  } else if (value.hasType<std::string>()) {
+    const auto stringValue = (std::string)value;
+    if (stringValue.back() == '%') {
+      auto tryValue = folly::tryTo<float>(
+          std::string_view(stringValue).substr(0, stringValue.length() - 1));
+      if (tryValue.hasValue()) {
+        return ValueUnit(tryValue.value(), UnitType::Percent);
+      }
+    }
+  }
+  return ValueUnit();
+};
```

Comment:
I think I still need to hook this up to OSS build more generally, but this code will eventually be replaced with a couple bits already landed.

`CSSValueVariant<CSSLength, CSSPercentage>`

Along with the equivalent `parseCSSProp` or `parseCSSComponentValue`.

Feel free not to use those yet though, since it's not yet hooked up more broadly.

Reply:
I don't think we'd want to use these here though @NickGerleman, since the code is fully layout independent? Or are you happy for these CSS utils to be exposed for non-layout purposes?

Reply:
I am planning to move the CSS values to be used in View/Image code as well, but am wanting to start with layout (mostly because it can happen before a chunk of work to change how styles in view managers work).